edge-modules/iotedge-diagnostics-dotnet/src/GetSocket.cs

```diff
@@ -21,6 +21,9 @@ public static string GetSocketResponse(string server, string endpoint)
             // Create a socket connection with the specified server and port.
             using (Socket socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.IP))
             {
+                socket.ReceiveTimeout = 5000;
```

Comment:
Have you confirmed that we get the desired behavior when a send or receive actually times out? For example, an error is logged in the diagnostics module (and ideally returned to `iotedge check`), but it doesn't cause the diagnostics module to crash?

Reply:
Yes, I added some error handler in the function and I could see the desired behavior.
![image](https://user-images.githubusercontent.com/6985754/118305817-5bec2580-b49d-11eb-8df8-8ed217460529.png)